README.md

```diff
@@ -16,57 +16,112 @@ Docker images can be found in Expedia Group's [dockerhub](https://hub.docker.com
-4. Hive events are picked up from the queue by Beekeeper using [Apiary Receiver](https://github.com/ExpediaGroup/apiary-extensions/tree/master/apiary-receivers).
-5. Beekeeper processes these messages and schedules orphaned paths or expired tables for deletion by adding them to a database.
-6. The scheduled paths or tables are deleted by Beekeeper after a configurable delay, the default is 3 days for paths and 30 days for tables (see [Hive table configuration](#hive-table-configuration) for more details.)
+3. Hive MetaStore events are emitted by the [Hive MetaStore Listener](https://github.com/ExpediaGroup/apiary-extensions/tree/master/apiary-metastore-listener) as a result of the operation.
+4. Hive events are picked up from the queue by Beekeeper using the [Apiary Receiver](https://github.com/ExpediaGroup/apiary-extensions/tree/master/apiary-receivers).
+5. Beekeeper processes these messages and schedules orphaned paths for deletion by adding them to a database.
+6. The scheduled paths are deleted by Beekeeper after a configurable delay, the default is 3 days (see [Hive table configuration](#hive-table-configuration) for more details.)
+
+## Time To Live, TTL 
+There is also a TTL property which, when added, will delete tables and their locations after a configurable delay. If no delay is specified the default is 30 days. 
+
+If the table is partitioned the cleanup delay will also apply to each partition that is added to the table. The table will only be dropped when there are no existing partitions. 
+
+End-to-end lifecycle example
+1. A Hive table is configured with the TTL parameter `beekeeper.remove.expired.data=true` (see [Hive table configuration](#hive-table-configuration) for more details).
+2. This Hive event is picked up from the queue by Beekeeper using the [Apiary Receiver](https://github.com/ExpediaGroup/apiary-extensions/tree/master/apiary-receivers), and the table is scheduled for cleanup with a configurable delay. 
+3. An operation is executed on the table which alters it in some way, (alter table, add partition, alter partition)
+4. These Hive events are once again picked up from the queue by Beekeeper using the Apiary receiver. Depending on the event, Beekeeper will do the following:
+    - `Alter table` - Creates a new entry in the database with the updated table info
+    - `Add partition` - The partition is scheduled to be deleted using the cleanup delay of the table 
+    - `Alter partition` - Creates a new entry in the database with the updated partition info
+5. The scheduled partitions, tables, and associated paths will be deleted by Beekeeper after the delay has passed.
+
+**TTL Caveats**
+
+Currently with the first release of the TTL there are the following issues:
+- If a table or partition is dropped by a user the related paths will become unreferenced and won’t be cleaned up. 
```

Comment:
Do you mean if they do a drop *before* the expiration happens?

Reply:
yes